### PR TITLE
completed auth

### PR DESCRIPTION
**Pull request by mohammadkhizerkhan**
added authslice
added private routes

### CODE REVIEW

File: src/features/auth/AuthSlice.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      CallToast("success", "Logged out Successfully");
+      return {
+        token: null,
+        user: null,
+      };
+    },
+  },
+  extraReducers: {
+    [signUpUser.pending]: (state) => {
+      console.log(state);
```
Comment (znjs) [inline src/features/auth/AuthSlice.js:59]: can remove the console's

File: src/features/auth/AuthSlice.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    [signUpUser.rejected]: (state, action) => {
+      state.error = action.payload;
+      CallToast(
+        "error",
+        action.payload.response.status === 422 &&
+          "User email is already present."
+      );
+    },
+    [loginUser.pending]: (state) => {
+      console.log(state);
```
Comment (znjs) [inline src/features/auth/AuthSlice.js:77]: can remove the console

**Review by znjs — APPROVED** [review]
- Good Pr
- Easy to understand code
- Feature Working as expected

**Review by saketkum11 — APPROVED** [review]
- UI Looks clean 
- much structure code and folder.

**Pull request merged**